### PR TITLE
feat: return full nextQuery object in paginated calls

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Combine a `nextPageRequest` from server with user provided custom options, this way user can just pass it to the next call as it is documented in the examples

https://github.com/googleapis/nodejs-spanner/blob/d835ba21489885f1fc092011d955c9ca5ef162ca/src/instance.ts#L1159-L1171

### CODE REVIEW

File: src/database.ts [hunk 1]
```diff
@@ -1336,7 +1336,7 @@ class Database extends GrpcServiceObject {
         reqOpts,
         gaxOpts,
       },
-      (err, sessions, ...args) => {
+      (err, sessions, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/database.ts:1339]: `nextPageRequest`

Comment (stephenplusplus) [inline src/database.ts:1339]: What is the format/properties of `nextPageRequest`?

File: src/index.ts [hunk 2]
```diff
@@ -534,7 +534,7 @@ class Spanner extends GrpcService {
         reqOpts,
         gaxOpts,
       },
-      (err, instances, ...args) => {
+      (err, instances, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/index.ts:537]: `nextPageRequest`*

File: src/index.ts [hunk 3]
```diff
@@ -737,7 +740,12 @@ class Spanner extends GrpcService {
         reqOpts,
         gaxOpts,
       },
-      callback
+      (err, instanceConfigs, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/index.ts:743]: `nextPageRequest`*

File: src/instance.ts [hunk 4]
```diff
@@ -351,15 +351,18 @@ class Instance extends common.GrpcServiceObject {
         reqOpts,
         gaxOpts,
       },
-      (err, backups, ...args) => {
+      (err, backups, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/instance.ts:354]: `nextPageRequest`*

File: src/instance.ts [hunk 5]
```diff
@@ -545,7 +548,13 @@ class Instance extends common.GrpcServiceObject {
         reqOpts,
         gaxOpts,
       },
-      callback
+      (err, operations, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/instance.ts:551]: `nextPageRequest`*

File: src/instance.ts [hunk 6]
```diff
@@ -663,7 +672,13 @@ class Instance extends common.GrpcServiceObject {
...
```
Comment (stephenplusplus) [inline src/instance.ts:675]: `nextPageRequest`*

File: src/instance.ts [hunk 7]
```diff
@@ -1221,7 +1236,7 @@ class Instance extends common.GrpcServiceObject {
         reqOpts,
         gaxOpts,
       },
-      (err, rowDatabases, ...args) => {
+      (err, rowDatabases, nexPageRequest, ...args) => {
```
Comment (stephenplusplus) [inline src/instance.ts:1239]: `nextPageRequest`*

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

File: src/database.ts (hunk 1 above)
Comment (AVaksman) [reply src/database.ts]: Thanks,
Fixed here and everywhere below.

Comment (AVaksman) [reply src/database.ts]: `nextPageRequest` comes in as `proto.~.IList<resource>Request` which is usually
```ts
parent/database;
pageSize;
pageToken;
//some request objects also have 
filter
```

**Review by stephenplusplus — APPROVED** [review]

**Comment by stephenplusplus** [comment]
@AVaksman should we merge, or do we need to wait for another review?

**Comment by stephenplusplus** [comment]
It seemed mergeable. Hopefully that wasn't a mistake!

**Pull request merged**